### PR TITLE
[ioctl] Build did geturi command line into new ioctl

### LINKED ISSUE #3292: [ioctl] build did geturi command line into new ioctl
### What would you like to be added
 - Step 1: create `didgeturi.go` and `didgeturi_test.go` files in `./ioctl/newcmd/did`. create this directory if not exist. 
 - Step 2: refactor  command  https://github.com/iotexproject/iotex-core/blob/master/ioctl/cmd/did/didgeturi.go in `didgeturi.go`
 - Step 3: add unit test in xx_test.go

#### Notes:
1. Use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to construct the Cobra command.
2. [Output package](https://github.com/iotexproject/iotex-core/blob/master/ioctl/output) is deprecated. `e.g.` `output.PrintError` could be replaced with `errors.Wrap`.
3. Global variables (config.ReadConfig  config.DefaultConfigFile etc.) have been wrapped into client interface. Please use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to access them if needed
4. The logic of new code should be similar to [the old one](https://github.com/iotexproject/iotex-core/blob/master/ioctl/cmd/did/didgeturi.go).
5. In each unit test, the results should be checked whether they are as the expected. 

#### Example:
1. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountinfo.go
2. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountdelete.go
3. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountcreateadd.go

### Why is this needed
Unit tests couldn't be added for the old ioctl command (./ioctl/cmd)

### PR DESCRIPTION
**Pull request by LuckyPigeon**
# Description

Refactor `didgeturi` command in new `ioctl`, with the following note.

* Use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to construct the Cobra command.
* Output package is deprecated, replace it with errors package.
* Replace fmt.Println with cmd.Println
* Refactor unit test to cover the modification.

Fixes #3292 

## Type of change
Please delete options that are not relevant.
- [x] Code refactor or improvement

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [x] TestNewDidGetUriCmd

**Test Configuration**:
- Firmware version: Windows 10 (WSL Version: Ubuntu 18.04)
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by LuckyPigeon** [comment]
@huof6829 
I use the same inter-multiple data hash of #3639, but seems it needs a different hash, could you check this one?

**Comment by huof6829** [comment]
> @huof6829 I use the same inter-multiple data hash of #3639, but seems it needs a different hash, could you check this one?

ok

**Comment by huof6829** [comment]
@LuckyPigeon  fixed.

File: ioctl/newcmd/did/didgeturi.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+				return errors.Wrap(err, "failed to get contract address")
+			}
+			addr, err := address.FromString(contract)
+			if err != nil {
+				return errors.Wrap(err, "invalid contract address")
+			}
+			abi, err := abi.JSON(strings.NewReader(DIDABI))
+			if err != nil {
+				return errors.Wrap(err, "failed to read abi")
+			}
```
Comment (CoderZhi) [inline ioctl/newcmd/did/didgeturi.go:58]: move to init(), and create a global parameter `_didABI`

**Review by CoderZhi — APPROVED** [review]

File: ioctl/newcmd/did/didgeturi_test.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+		cmd := NewDidGetURICmd(client)
+		_, err := util.ExecuteCmd(cmd, accAddr, "did:io:0x11111111111111111")
+		require.Contains(err.Error(), expectedErr.Error())
+	})
+
+	t.Run("failed to read contract", func(t *testing.T) {
+		expectedErr := errors.New("failed to read contract")
+		apiServiceClient.EXPECT().ReadContract(gomock.Any(), gomock.Any()).Return(nil, expectedErr)
+		cmd := NewDidGetURICmd(client)
+		_, err := util.ExecuteCmd(cmd, accAddr, payload)
```
Comment (huof6829) [inline ioctl/newcmd/did/didgeturi_test.go:75]: last argument is DID

File: ioctl/newcmd/did/didgeturi_test.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+		require.Contains(err.Error(), expectedErr.Error())
+	})
+
+	t.Run("invalid contract address", func(t *testing.T) {
+		expectedErr := errors.New("invalid contract address")
+		client.EXPECT().Address(gomock.Any()).Return("test", nil)
+		cmd := NewDidGetURICmd(client)
+		_, err := util.ExecuteCmd(cmd, "test", payload)
```
Comment (huof6829) [inline ioctl/newcmd/did/didgeturi_test.go:83]: last argument is DID

File: ioctl/newcmd/did/didgeturi_test.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+		require.Contains(err.Error(), expectedErr.Error())
+	})
+
+	t.Run("failed to get contract address", func(t *testing.T) {
+		expectedErr := errors.New("failed to get contract address")
+		client.EXPECT().Address(gomock.Any()).Return("", expectedErr)
+		cmd := NewDidGetURICmd(client)
+		_, err := util.ExecuteCmd(cmd, "test", payload)
```
Comment (huof6829) [inline ioctl/newcmd/did/didgeturi_test.go:91]: last argument is DID

**Review by huof6829 — APPROVED** [review]

**Comment by LuckyPigeon** [comment]
> @LuckyPigeon fixed.

Thanks for the help.

**Pull request merged**